### PR TITLE
fix wrong requested CPU count when node has finished jobs

### LINKED ISSUE #65207: Wrong requested CPU count when node has finished jobs
/kind bug

**What happened**:
In nodes where there are only 2 CPU I can see requested values higher than 2.
It seems to be because of successful jobs that are kept (jobs managed by a cronjob).
This is making autoscaler scale more and more nodes, as those where job have been scheduled remain "full".

This is happening in GKE, but its likely to happen in any k8s cluster:

image

image

You can see that the SUM of all POD CPU requested matches the node CPU request.

**What you expected to happen**:
When a job is successful it shouldn't count agains the requested cpu count.

**How to reproduce it (as minimally and precisely as possible)**:
Ejecute a CronJob and wait until it finishes. Wait for more executions.

**Anything else we need to know?**:
As a workaround, it is possible to set the CronJob `.spec.successfulJobsHistoryLimit` to 0, so successful jobs are not kept.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.9.7
- Cloud provider or hardware configuration: GKE
- OS (e.g. from /etc/os-release): v1.9.6-gke.1 (COS)
- Kernel (e.g. `uname -a`): v1.9.6-gke.1 (COS)

### PR DESCRIPTION
**Pull request by wgliang**
**What this PR does / why we need it**:
/kind bug

Set the CronJob spec.SuccessfulJobsHistoryLimit to 0, so successful jobs are not kept.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #65207

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controller.go [hunk 1]
```diff
@@ -156,6 +156,8 @@ func cleanupFinishedJobs(sj *batchv1beta1.CronJob, js []batchv1.Job, jc jobContr
 			pc,
 			*sj.Spec.SuccessfulJobsHistoryLimit,
 			recorder)
+		sj.Spec.SuccessfulJobsHistoryLimit = new(int32)
```
Comment (yastij) [inline pkg/controller/cronjob/cronjob_controller.go]: This should also be done for `FailedJobsHistoryLimit`

Comment (wgliang) [reply pkg/controller/cronjob/cronjob_controller.go]: Done. Thanks for your review.

**Review by yastij — CHANGES_REQUESTED** [review]
@wgliang - seems that won’t fix the issue, If setting this to 0, when it was a non-zero value before will cause deletion of all jobs when reconciling again. The solution would be to not count finished jobs into CPU requests. cc @soltysh

File: pkg/controller/cronjob/cronjob_controller.go (hunk 1 above)
Comment (soltysh) [reply pkg/controller/cronjob/cronjob_controller.go]: That is not correct, you're changing history limits after cleaning, why?

**Review by soltysh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**